lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/classify/MultiMADClassifier.java

```diff
@@ -0,0 +1,292 @@
+            assert (MAD != 0);
+        }
+
+        Stats stats = new Stats(median, MAD * MAD_TO_ZSCORE_COEFFICIENT);
+        return stats;
+    }
+
+    @Override
+    public DataFrame getResults() {
+        return output;
+    }
+
+    // Parameter Getters and Setters
+    public double getPercentile() {
+        return percentile;
+    }
+
+    public double getZscore() {
+        return zscore;
+    }
+
+    public double getSamplingRate() {
+        return samplingRate;
+    }
+
+    public boolean isIncludeHigh() {
+        return includeHigh;
+    }
+
+    public boolean isIncludeLow() {
+        return includeLow;
+    }
+
+    public boolean isUseParallel() {
+        return useParallel;
+    }
+
+    public boolean isUseReservoirSampling() {
+        return useReservoirSampling;
+    }
+
+    public List<String> getColumnNames() {
+        return columnNames;
+    }
+
+    public String getOutputColumnSuffix() {
+        return outputColumnSuffix;
+    }
+
+    /**
+     * @param percentile Approximate percent of data to classify as outlier on each end of
+     * spectrum (i.e., a percentile of 5 means that the top 5% and bottom 5% are outliers)
+     * @return this
+     */
+    public MultiMADClassifier setPercentile(double percentile) {
+        this.percentile = percentile;
+        // https://en.wikipedia.org/wiki/Normal_distribution#Quantile_function
+        this.zscore = Math.sqrt(2) * Erf.erfcInv(2.0*percentile/100.0);
+        return this;
+    }
+
+    /**
+     * @param zscore Z-score above/below which data is classified as outlier
+     * @return this
+     */
+    public MultiMADClassifier setZscore(double zscore) {
+        this.zscore = zscore;
+        return this;
+    }
+
+    /**
+     * @param samplingRate Rate to sample elements used to calculate median and MAD, must be
+     * greater than 0 and at most 1 (i.e., use all elements without sampling)
+     * @return this
+     */
+    public MultiMADClassifier setSamplingRate(double samplingRate) {
+        this.samplingRate = samplingRate;
+        return this;
+    }
+
+    /**
+     * @param includeHigh Whether to count high points as outliers.
+     * @return this
+     */
+    public MultiMADClassifier setIncludeHigh(boolean includeHigh) {
+        this.includeHigh = includeHigh;
+        return this;
+    }
+    
+    /**
+     * @param includeLow Whether to count low points as outliers
+     * @return this
+     */
+    public MultiMADClassifier setIncludeLow(boolean includeLow) {
+        this.includeLow = includeLow;
+        return this;
+    }
+
+    /**
+     * @param useParallel Whether to use parallelization, can be beneficial for large datasets
+     * @return this
+     */
+    public MultiMADClassifier setUseParallel(boolean useParallel) {
+        this.useParallel = useParallel;
+        return this;
+    }
+
+    /**
+     * @param use ReservoirSampling Whether to sample using reservoir sampling. Otherwise,
```

Comment:
Since you took out the magic number in the original code, I would probably add one or two more sentences here on when it's good to use reservoir sampling VS Fisher-Yates.